src/rgw/services/svc_config_key_rados.cc

```diff
@@ -0,0 +1,16 @@
+
+#include "svc_rados.h"
+#include "svc_config_key_rados.h"
+
+int RGWSI_ConfigKey_RADOS::get(const string& key, bufferlist *result)
+{
+  string cmd =
+    "{"
+      "\"prefix\": \"config-key get\", "
+      "\"key\": \"" + key + "\""
+    "}";
```

Comment:
While we're at it. For completeness sake, should be make this also able to get stuff out of the config database, not just the config-key database?

Maybe we can't call it ConfigKey in that case.

Maybe it needs to be MonConfig::get and MonConfig::getkey or something(or just GSWSI_Config::)? If you look into the cephadm rgw code, they set the realm and zone using `config set` (https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/module.py#L2089-L2100). So we need to plumb RGW to also check for not just `config-key get` but also `config get`. 

This will take the who into account too and not have to worry about different keys for different RGW instances. 

Just a thought

Reply:
oh, looking closer, can we just access the `config get` from config-key using key:
`config/client.rgw.<realm>.<zone>` but it's the processing the scope of the config which is, I'm assuming, already handled by the `mon config`, ie, the who. For example, do I need to check `config/client/option` or `config/client.rgw,realm/option` or `config/client.rgw.realm.zone` etc

That'll be alot more requests to the mon asking about existence of options then just a single call to `mon config get`

Reply:
@matthewoliver I'm not sure I quite follow you, but note that `config get` won't apply the defaults on `rgw_frontends`

Reply:
Sorry, I'm probably confusing things. The problem is outside of rgw_frontends. The current version of the rgw code in cephadm sets the `rgw_realm` and `rgw_zone` options via `config set`. But not sure RGW currently can or does get these values from the `config get`. So am guessing will always default to `default` for both.

I know that is out of scope of what your trying to do here. But was raising the suggestion of extending the plumbing you've done to include `config get` in the hopes that it means we don't need to dump the realm and zone in the ceph.conf in the container to really get it to work.

NOTE: I'm assuming here that in fact RGW currently isn't talking to `config get` as I can't see any mon_commands that do that. And pretty sure the ConfigProxy object doesn't.

Reply:
You can configure these params (realm, zonegroup, zone) via the ceph config commands.